flang/runtime/dot-product.cpp

```diff
@@ -83,8 +89,11 @@ static inline CppTypeFor<RCAT, RKIND> DoDotProduct(
       AccumType accum{};
       if constexpr (RCAT == TypeCategory::Complex) {
         for (SubscriptValue j{0}; j < n; ++j) {
-          accum += std::conj(static_cast<AccumType>(*xp++)) *
+          // std::conj() may instantiate its argument twice,
```

Comment:
I don't understand this point.  Unless `std::conj` is a macro, how could it possibly evaluate its argument more than once?

Reply:
The issue shows up as a warning `multiple unsequenced modifications to 'xp'`, when compiling the runtime as OpenMP offload with clang.  I think it might be a bug in clang, because `std::conj` is not a macro.  I can change the comment to something like `avoid spurious warning when compiling with clang` to avoid the confusion about multiple instantiation of the argument.

Reply:
No, this is fine, just please extend the comment to mention that this new code is a work-around to what is perhaps a bug in clang, not something that a C++ programmer should have to worry about in general.